### ISSUE TITLE
running baseline experiments

### ISSUE THREAD

**Issue by ekinakyurek** [yafuly/CoGnition#3]
Hi @yafuly,

I am trying run baseline fairseq models. From the fairseq logs, it seems like I am getting 61.2 BLEU for validation set which doesn't match with the scores in README.

After training finished, I tried to run `infer-one-click-run.sh` to get cg-test results as well. However, I got a 'dataset not found' error from fairseq.

In the inference script, I set relative paths as in below:
```
PATH_TO_FAIRSEQ_DATABIN="CoGnition/data/processed/bin"
PATH_TO_MODEL="exp/CoGnition/checkpoint_best.pt"
PATH_TO_FAIRSEQ_CLI="fairseq/fairseq_cli/"
PATH_TO_COGNITION_EVAL="CoGnition/eval/"
```

The files under `CoGnition/data/processed/bin` are exactly same as in [here](https://github.com/yafuly/CoGnition/tree/master/data/processed/bin)

So, I wonder if there is a missing file or do I set paths wrong?

After a little search, I found that if I remove `--dataset-impl "raw"` flag from the script, the fairseq code works. But then I got error in  `_reorder.py` part of the script.

Could you guide me to solve these issues?

Thanks
Ekin

**Comment by ekinakyurek** [comment]
Update:

I found that if I copy three files inside the [bin folder](https://github.com/yafuly/CoGnition/tree/master/data/processed/bin), the infer script works without changing any other thing:

```
cp data/cg-test/cg-test.compound data/processed/bin/
cp data/processed/cg-test.zh data/processed/bin/cg-test.en-zh.zh
cp data/processed/cg-test.en data/processed/bin/cg-test.en-zh.en
```

But, I got cg-test BLEU score as 46.5. So, my final BLEU scores are:

valid: 61.2
cg-test: 46.5

I assume that I do have an eval mismatch, your help is appreciated again

Thanks

**Comment by yafuly** [comment]
Hi Ekin,

We didn't report scores on the validation set. The scores on the Github table were obtained by training models under different seeds, choosing the best checkpoint based on the valid set, then conducting inference on test sets and averaging their scores.

Based on your recent updates, it seems you are able to conduct inference successfully but get a much lower BLEU score on the cg-test. Did you use the same sacrebleu command line in the `infer-one-click-run.sh'? It might be some errors in computing sacrebleu scores.

Thanks,
Yafu

**Comment by ekinakyurek** [comment]
Seems like my copied ref file has BPE tokens in it, so I added `sed` to line 21 in infer-one-click-run.sh to remove them.

Now the results are similar to README

Thanks
